src/main/java/seedu/duke/Duke.java

```diff
@@ -12,14 +12,14 @@
     private TextUi textUi;
     private String contactFilePath;
     private Storage storage;
-    private Parser parser;
+    private MainParser parser;
     private ContactList contactList;
 
     public Duke(String contactFilePath) {
         this.textUi = new TextUi();
         this.contactFilePath = contactFilePath;
         this.storage = new Storage(contactFilePath);
-        this.parser = new Parser();
+        this.parser = new MainParser();
```

Comment:
Minor nitpick, but maybe this.parser could be changed to this.mainparser for consistency?